### DISCUSSION TITLE
Issue while the logon scripts are running inside the HCIbox

### DISCUSSION THREAD

**Discussion by ullasanand13** [microsoft/azure_arc#1795] [General]
Hi GIT team,

"HCIBox-Client" has got deployed. However, after login to the HCIbox the logon scripts have failed due to the below error. I know this is a permissions issue. But it is trying to connect to our default subscription. How to change the subscription where the resources have to be deployed?

Message: The client 'ullas.anand@kennametal.com' with object id 'f0851858-ef81-48ef-b258-b865c3e268ce' does not have authorization to perform action 'Microsoft.GuestConfiguration/register/action' over scope '/subscriptions/7e62512b-1073-46d1-b75c-c507c9cc7eb2' or the scope is invalid. If access was recently granted, please refresh your credentials.
(AuthorizationFailed) The client 'ullas.anand@kennametal.com' with object id 'f0851858-ef81-48ef-b258-b865c3e268ce' does not have authorization to perform action 'Microsoft.Kubernetes/register/action' over scope '/subscriptions/7e62512b-1073-46d1-b75c-c507c9cc7eb2' or the scope is invalid. If access was recently granted, please refresh your credentials.
Code: AuthorizationFailed

**Comment by FallenHoot** [comment]
This simply means that your user doesn't have access to that subscription to perform those actions. Verify your RBAC on the target subscription. Otherwise, change/set the subscription, you can use the following command.

`az account set --subscription "00000000-0000-0000-0000-000000000000"` 

Then recreate your resource group so that it is in the correct subscription.

**Comment by ullasanand13** [comment]
Thank you. I was able to overcome this issue by giving contributor role to the service principal. 

Regards,
Ullas
